projects/frontend/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -9,33 +12,146 @@ import { environment } from 'projects/frontend/src/environments/environment';
 })
 export class DashboardComponent implements OnInit {
 
-  private apiUrl: string;
-  constructor(private http: HttpClient) {
-    this.apiUrl = environment.API_URL;
-  }
+  constructor(
+    private authService: AuthService,
+    private taskService: TaskService,
+    private formBuilder: FormBuilder,
+    private toast: ToastrService
+    ) {}
 
   tasks;
-  all;
-  completed;
-  ongoing;
-  upcoming;
-  percentComp;
+  currentUser;
+  addForm: FormGroup;
+  updateForm: FormGroup;
+  loading = false;
+  updating = false;
+  adding = false;
+  task;
+  eTask;
 
   ngOnInit(): void {
+    this.getUser();
+    this.addForm = this.formBuilder.group({
+      task: ['', [Validators.required]],
+      priority: ['', [Validators.required]],
+      status: ['', [Validators.required]]
+    });
+    this.updateForm = this.formBuilder.group({
+      task: ['', [Validators.required]],
+      priority: ['', [Validators.required]],
+      status: ['', [Validators.required]]
+    });
+  }
+
+  get af() { return this.addForm.controls; }
+  get uf() { return this.updateForm.controls; }
+
+  getUser() {
+    this.currentUser = this.authService.getLoggedInUser();
     this.listTasks();
   }
 
-  // tslint:disable-next-line:typedef
-  listTasks() {
-    const url = this.apiUrl + '/tasks';
-    this.http.get(url).subscribe(res => {
-      this.tasks = res;
+  viewTask(id) {
+    this.taskService.getTask(id).subscribe(
+      data => {
+        this.task = data;
     });
-    this.loadStatistics(this.tasks);
   }
 
-  // tslint:disable-next-line:typedef
-  loadStatistics(tasks){
+  isSelectedPriority(p){
+    if (this.eTask.priority === p){
+      // console.log(p);
+      return 'selected';
+    } else {
+      return '';
+    }
+  }
+  isSelectedStatus(s){
+    if (this.eTask.status === s){
+      // console.log(s);
+      return 'selected';
+    } else {
+      return '';
+    }
+  }
+
+  updateTask() {
+    this.loading = true;
+
+    const editedTask = new Task(this.uf.task.value, this.currentUser.id, this.uf.priority.value, this.uf.status.value);
+
+    if (this.updateForm.invalid) {
+      return;
+    }
+
+    this.updating = true;
+
+    console.log('calling api');
+    this.taskService.updateTask(this.eTask.id , editedTask).subscribe(
+      data => {
+        console.log('updated', data);
+        this.listTasks();
+        this.toast.success('added edited!');
+      }, error => {
+        // console.log(error);
+        this.toast.error(error);
+      }
+    );
+  }
+
+  editTask(id) {
```

Comment:
in editTask instead of passing id , you can pass "task" object, so that you can avoid a backend rest api call.